server/app.js

```diff
@@ -27,9 +32,37 @@ const io = socketio(server, {
     origin: "*",
   },
 });
+io.use(ioCookieParser());
+io.use((socket, next) => {
+  const token = socket.request.cookies["token"];
+  if (token) {
+    try {
+      const decoded = jwt.verify(token, process.env.JWT_SECRET);
+      user = decoded;
```

Comment:
I'm not sure if this variable `user` has a declaration, and decoded should contain more than "user" object or so. So you need to extract the field you're interested in from the `decoded` object. I also cannot see where `user` is used here. I think you might want to add the decoded user to the `socket` object so it becomes available in the next middleware.

Reply:
yes indeed I was thinking like this, but after i figure out it is sufficient to use ids